### ISSUE TITLE
Unable to create record

### ISSUE THREAD

**Issue by leandrolopes13** [twilio/twilio-php#305]
Version: Next-gen 
```
function message(){
        $this->client = new Twilio\Rest\Client($this->CI->config->item("twilio_sid"), $this->CI->config->item("twilio_token"));
        try{
            $message = $this->client->api->v2010->account->messages->create(
                'xxx', // From a valid Twilio number
                'xxx',
                array(
                    'body' => 'Test message'
                )
            );
            var_dump($message);
            return $message;
        }catch(Exception $e){
            echo $e->getMessage();
            return FALSE;
        }
    }
```
Generate an exception: Unable to create record.

**Comment by rockoo** [comment]
I get the same result on occasion. I.e. When you send in batch some numbers will get 404 Unable to create record... No information on twilio side on this

**Comment by caponica** [comment]
I get the same remarkably helpful "unable to create record" error when trying to use the test account:

    $twilioClient = new Client($twilioSid, $twilioToken);
    $message = $twilioClient->getAccount()->calls->create(
      '+15005550003', '+15005550006', [
        'Url' => 'http://demo.twilio.com/docs/voice.xml'
      ]
    );

I'm using the test credentials from my account for the Sid and Token.

As far as I can tell from [the docs](https://www.twilio.com/docs/api/rest/test-credentials) this should give me a 21215 error but it just says "Unable to create record". Using a valid "to" number, or any of the magic numbers 15005550001-4 just gives the same error.

Am I doing something wrong or is this just the wayTwilio 'works'?

**Comment by caponica** [comment]
OK, I've figured out why the exception message is so unhelpful - it is not passing the Twilio message/code through to the Exception. This is fixed in #311

**Comment by caponica** [comment]
Oh, and I figured out the problem in my code above. It should be 'url', not 'Url'.

How about some proper domain objects for options with code-completion hints instead of arrays with literal keys? Or at least some class constants for the key names?

**Comment by apoca** [comment]
I have the same error:

$client = new TwilioClient($this->config['twilio']['sid'], $this->config['twilio']['token']);
                $message = $client->api->v2010->account->messages->create(
                    $this->config['twilio']['number'], // From a valid Twilio number
                    $params['phone'], // Text this number
                    [
                        'Body' => sprintf($this->translator->translate(": %s"), $randCode)
                    ]
                );

**Comment by carlosdp** [comment]
@apoca yea, just like @caponica you are using upper-camelcased keys when the library expects lower-camelcased keys, as that is the more common PHP convention.

The next release candidate will add a pre-processing step to array key input that will let the library accept keys case-insensitive so that you can use upper or lower cased keys.

**Comment by ihumanable** [comment]
Case-insensitive key handling added in https://github.com/twilio/twilio-php/commit/c9be682625cc82c2b5aee96dd1368e1f3c965b76

Tested in https://github.com/twilio/twilio-php/commit/3b42c0925761327da5972c0022c2fa19010b3372

Released as 5.0.0-RC19

I'm happy to consider more documentation around this, it doesn't seem that the PHP Community has fully agreed to an array documenting standard.  If there is one I'm missing please let me know.

**Comment by ihumanable** [comment]
@apoca @caponica @leandrolopes13 

Thank you for providing feedback, based on the issues you encountered we are adding `Option` classes to the helper library to provide better editor support.  A PR has been opened here https://github.com/twilio/twilio-php/pull/364.  Any additional feedback would be appreciated.
